### ISSUE TITLE
Plugin does not support aws_appmesh_virtual_gateway and aws_appmesh_gateway_route

### ISSUE THREAD

**Issue by naishe** [VladRassokhin/intellij-hcl#323]
The plugin does not seem to support `aws_appmesh_virtual_gateway` and `aws_appmesh_gateway_route`.

1. It does not suggest these modules in autocomplete
    <img width="668" alt="Autocomplete does not work" src="https://user-images.githubusercontent.com/845424/106553429-64000500-653f-11eb-9bb6-de37a2b34d19.png">
2. It shows unsupported block warning on valid blocks
    <img width="523" alt="Wrong warnings" src="https://user-images.githubusercontent.com/845424/106553523-8c87ff00-653f-11eb-9af3-66759c89ead5.png">

The examples are taken directly from https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appmesh_virtual_gateway (latest, at the time of posting this issue the version was mentioned `3.26.0`)
The details are as follows:

### Prerequisites
 - [x] Ensure you have latest version of plugin installed
     <img width="327" alt="version 0.7.10" src="https://user-images.githubusercontent.com/845424/106550053-58f5a680-6538-11eb-8f12-b44ea7f6861e.png">
 - [x] Search for possible issue duplicates

### Installation details
 - [x] IDE version (`Help->About->Copy to Clipboard`)
    ```
    PyCharm 2019.3.5 (Community Edition)
    Build #PC-193.7288.30, built on May 7, 2020
    Runtime version: 11.0.6+8-b520.66 x86_64
    VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
    macOS 10.16
    GC: ParNew, ConcurrentMarkSweep
    Memory: 1981M
    Cores: 8
    Registry: 
    Non-Bundled Plugins: org.intellij.plugins.hcl
    ```
 - [x] intellij-hcl plugin version (`Settings->Plugins`)
     Hashicorp Terraform / HCL Language Support, JetBrains, 0.7.10
 - [x] Terraform version (`terraform -v`)
     terraform -v
    Terraform v0.14.3

### Terraform Configuration Files
```hcl
resource "aws_appmesh_virtual_gateway" "example" {
  name      = "example-virtual-gateway"
  mesh_name = "example-service-mesh"

  spec {
    listener {
      port_mapping {
        port     = 8080
        protocol = "http"
      }
    }
  }

  tags = {
    Environment = "test"
  }
}
```

### Exception
It shows `unknown block type` for `spec`, `listener`, and `port_mapping`
```text
Unknown block type spec
```

### Expected Behavior
The autocomplete should work
The warnings should not be shown

### Actual Behavior
The autocomplete does not work
Warnings are shown on valid blocks

### Steps to Reproduce
Create a configuration using examples given in Terraform documentation for [appmesh_virtual_gateway](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appmesh_virtual_gateway) and [appmesh_gateway_route](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appmesh_gateway_route)

Observe the following:
1. The auto-completion does not suggest these resources
2. Valid blocks show warning as specified above

**Comment by naishe** [comment]
Continuing the thread, there are more places where the plugin does not work as intended. For example, this code exactly taken from the documentation has two issues:

```hcl
resource "aws_appmesh_virtual_service" "admin_service" {
  mesh_name = aws_appmesh_mesh.mesh.id
  name = local.admin_service_dns
  spec {
    provider {
      virtual_node {
        virtual_node_name = aws_appmesh_virtual_node.admin_node.name
      }
    }
  }
}
```

1. `provider` block does not come in autocomplete (nor any of the attributes within this tag)
2. It wrongly shows it as error, as illustrated in the image below
    <img width="703" alt="Screen Shot 2021-02-02 at 4 42 55 PM" src="https://user-images.githubusercontent.com/845424/106593098-62056880-6576-11eb-8691-012cdc71768d.png">

**Comment by naishe** [comment]
@VladRassokhin sorry to bother, any idea?

**Comment by haidaraM** [comment]
For different providers resources, check this: https://github.com/VladRassokhin/intellij-hcl#terraform-external-metadata
You need to build the schema manually.

**Comment by naishe** [comment]
> For different providers resources, check this: https://github.com/VladRassokhin/intellij-hcl#terraform-external-metadata
> You need to build the schema manually.

Should it not work out of the box? This is not external meta data. This resource and properties are documented [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appmesh_virtual_gateway) so, like any other AWS resource the properties should auto complete and not show a warning.

This project being closed sourced, I can't even pry open and see why is this the case.

**Comment by haidaraM** [comment]
The providers schemas is embeded with the plugin as the time the plugin is released. Since there is no new release of the plugin, if you want to have autocompletion for the new resources, you will need to do get the schemas manually or maybe use Terraform language server : https://github.com/hashicorp/terraform-ls/blob/main/docs/USAGE.md#intellij-ide

Otherwise, check my fork of the metadata that I update from time to time: https://github.com/haidaraM/terraform-metadata/tree/feat/updateProviders. You have the `aws_appmesh_virtual_gateway` there :-) Just follow the instructions and should work

**Comment by VladRassokhin** [comment]
Partially fixed in 0.7.11
Fix for `aws_appmesh_virtual_service.spec.provider.virtual_node` will be in next update (0.7.13)

**Comment by VladRassokhin** [comment]
Fixed in 0.7.13
